### PR TITLE
Fix stale notifications on mobile

### PR DESCRIPTION
**Pull request by Robinnnnn**
### Summary of Changes

- [x] refetch notifications list properly when visited
- [x] ensure blue dot appears on app foreground
- [x] periodically refetch blue dot every 10s
- [x] ensure admire count gets un-stale

cool helper functions: `useEffectOnAppForeground`, `useIntervalEffectOnAppForeground`

### Demo or Before/After Pics

https://www.loom.com/share/c0da1d2b7024400c9b76d00e75560021?sid=939c250e-4eaa-4225-8490-0ec878212c03

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/mobile/src/utils/useIntervalEffectOnAppForeground.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
+import { AppState, AppStateStatus } from 'react-native';
+
+export function useIntervalEffectOnAppForeground(callback: () => void) {
```
Comment (kaitoo1) [inline apps/mobile/src/utils/useIntervalEffectOnAppForeground.ts:4]: 😍😍😍😍

**Review by kaitoo1 — APPROVED** [review]
